ff/src/fields/mod.rs

```diff
@@ -268,6 +268,102 @@ pub trait Field:
     }
 }
 
+/// Fields that have a cyclotomic multiplicative subgroup, and which can
+/// leverage efficient inversion and squaring algorithms for elements in this subgroup.
+/// If a field has multiplicative order p^d - 1, the cyclotomic subgroups refer to subgroups of order φ_n(p),
+/// for any n < d, where φ_n is the [nth cyclotomic polynomial](https://en.wikipedia.org/wiki/Cyclotomic_polynomial)
+pub trait CyclotomicField: Field {
```

Comment:
Thoughts on naming the trait `CyclotomicSubgroup`, to better highlight this is improved arithmetic for points that are in a subgroup, not the entire field?

Reply:
That's a good idea. I'd like to find to do that while avoiding confusion with the rest of the changes coming in from #294, which re-introduce an _additive_ group abstraction (while this is a _multiplicative_ group abstraction)

Reply:
Renamed to `CyclotomicMultSubgroup`, and added a note specifying that this is different from `ark_ec::Group`.